src/plugins/upf/upf_api.c

```diff
@@ -96,9 +99,11 @@ vl_api_upf_app_add_del_t_handler (vl_api_upf_app_add_del_t * mp)
   vl_api_upf_app_add_del_reply_t *rmp = NULL;
   upf_main_t *sm = &upf_main;
   int rv = 0;
+  u8 *name = format(0, "%s", mp->name);
```

Comment:
there is a simpler vec function for this

Reply:
Same, I've basically copied that from other plugins. I'm not very well-versed in VPP's clib atm, will look for a simpler way to do it

Reply:
Actually, I can see `vec_validate_init_c_string()` macro, which is used just once, and some code just uses `vac_validate()`, neither being more concise than this `format()` call, performance issues from calling `format()` not being relevant to these relatively rare API calls, so I'd appreciate any pointers concerning what function should I use. Re this `format()` trick, it seems to be used quite a bit in the VPP source:
```
git grep 'format (0, "%s"' | wc -l
      87
```

Reply:
... ok, I see, will fix that

Reply:
yeah, seems that lots of people just copy from each other. Leave it in if you want. The overhead of the format string parsing is probably not worth it.